### ISSUE TITLE
Experiment Result CCT_7

### ISSUE THREAD

**Issue by StephenEkaputra** [SHI-Labs/Compact-Transformers#10]
Hi,

Thank you for the wonderful paper.
I have trained the CCT_7/3x1 by default and however, the result is differ from the paper.
CIFAR10: 93.67
CIFAR100: 73.15
I've tried it many times and It never reaches 94.72% for CIFAR10 and 76.67% for CIFAR100.
Could you please help me what makes it differ?

Thanks.

**Comment by alihassanijr** [comment]
Hi,

Could you please share more details on the training jobs, specifically the hyperparameters.

**Comment by StephenEkaputra** [comment]
Hi,

I used the following settings:
learning rate 5e-4
weight decay 3e-2
batch size 64
warm up 5 epochs
epoch 200
pos emb = default 'learnable'

**Comment by alihassanijr** [comment]
That's strange. Let me try that and get back to you.

**Comment by StephenEkaputra** [comment]
Thank you for your attention..
Looking forward to hearing from you

**Comment by stevenwalton** [comment]
Hey Stephen, what kind of graphics cards are you using? I don't want to rule out silicon lottery. Can you try increasing the maximum number of epochs? And are you reporting the final result or the best result? This stuff is unfortunately always a bit tricky (I've had plenty of frustrations trying to replicate papers too. It's never as easy as it seems)

**Comment by StephenEkaputra** [comment]
Hi Steven..

I'm using RTX3090 and those results are the best result..
I'm trying to increase epochs to 300 and will inform you the result..
I wish the best for you.. Good luck!

**Comment by alihassanijr** [comment]
Hi again @StephenEkaputra 
Thank you for bringing this to our attention. We seem to have mistakenly removed one augmentation from our training script, which was added back. We just trained the same model (although we use 10 warmup epochs in the original experiment, shouldn't matter that much but I'd start with that for CIFAR-10) with the latest commit and it works out fine.
I apologize for the inconvenience.

**Comment by StephenEkaputra** [comment]
Hi,
Thank you for the update.. It's very helpful..
After add the augmentation with random crop, the accuracy of CCT_7/3x1 improves to:
CIFAR10: 93.77%
CIFAR100:  74.91%

![image](https://user-images.githubusercontent.com/31305502/121816418-12fedb00-ccae-11eb-9382-fe0a911abe61.png)

I use the following settings:
learning rate 5e-4
weight decay 3e-2
batch size 64
warm up 10 epochs
epoch 200
pos emb = default 'learnable'

However, we can't reach the accuracy of 94.72% for CIFAR10 and 76.67% for CIFAR100.
If you don't mind, can you share the weight .pt model (CCT_7/3x1) trained on CIFAR10 and CIFAR100 that reaches the accuracy according to the paper?

Thank you for your kind attention.

**Comment by alihassanijr** [comment]
Hi,
This is very strange, because when we tried it on CIFAR-10, we ran it 8 times and all 8 were above 94%.
I've uploaded one checkpoint here that you can try: [cct7-3x1_cifar10.pth.zip](https://github.com/SHI-Labs/Compact-Transformers/files/6644379/cct7-3x1_cifar10.pth.zip). You can use the evaluate script (just pushed it to the repo):
`python evaluate.py --checkpoint-path cct7-3x1_cifar10.pth /path/to/cifar10`

Let me know how it goes.

**Comment by StephenEkaputra** [comment]
Hi,

Thank you for the update..
The checkpoint you gave me is actually correct.. the accuracy of CIFAR10 I got is 94.78%
It is very strange.. I wonder if the graphic card affects the performance?
May I also try the CIFAR100 checkpoint if possible?

**Comment by alihassanijr** [comment]
Hi,

Yes, it sounds strange, because we ran multiple separate runs and I don't think any stopped below 94.
We also just added the `--dataset` argument to our training and evaluation scripts, which supports CIFAR-100 as well. The checkpoint to our best CIFAR-100 run is also updated in README.
You can also download it [here](https://github.com/SHI-Labs/Compact-Transformers/files/6649937/cct7-3x1_cifar100.pth.zip). It's actually at **77.05 %** top-1 accuracy now.

**Comment by alihassanijr** [comment]
I'm closing the issue now, if you still have questions feel free to open it back up.
